presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -3128,6 +3128,15 @@ public void testIn()
         assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (1, 2, 3)");
         assertQuery("SELECT orderkey FROM orders WHERE orderkey IN (1.5, 2.3)");
         assertQuery("SELECT orderkey FROM orders WHERE totalprice IN (1, 2, 3)");
+        assertQuery("SELECT x FROM (values 3, 100) t(x) WHERE x IN (2147483649)", "SELECT * WHERE false");
+        assertQuery("SELECT x FROM (values 100, 3) t(x) WHERE x IN (2147483649)", "SELECT * WHERE false");
+        assertQuery("SELECT x FROM (values 3, 100) t(x) WHERE x IN (3, cast(rand() * 12345678910 as bigint))", "SELECT 3");
+        assertQuery("SELECT x FROM (values 3, 100, 2147483648, 2147483649, 2147483650) t(x) WHERE x IN (2147483648, 2147483650)", "SELECT * From (values 2147483648, 2147483650)");
+        assertQuery("SELECT x FROM (values 3, 100, 2147483648, 2147483649, 2147483650) t(x) WHERE x IN (3, 4, 2147483648, 2147483650)", "SELECT * From (values 3, 2147483648, 2147483650)");
+        assertQuery("SELECT x FROM (values 1, 2, 3) t(x) WHERE x IN (1 + cast(rand() < 0 as bigint), 2 + cast(rand() < 0 as bigint))", "SELECT * From (values 1, 2)");
+        assertQuery("SELECT x FROM (values 1, 2, 3, 4) t(x) WHERE x IN (1 + cast(rand() < 0 as bigint), 2 + cast(rand() < 0 as bigint), 4)", "SELECT * From (values 1, 2, 4)");
+        assertQuery("SELECT x FROM (values 1, 2, 3, 4) t(x) WHERE x IN (4, 2 + cast(rand() < 0 as bigint), 1 + cast(rand() < 0 as bigint))", "SELECT * From (values 1, 2, 4)");
```

Comment:
Non-constant expression does not go in the list of lookupswitch anyway. Please add out-of-order tests that doesn't have ` + cast(rand() < 0 as bigint)`.